src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -417,6 +411,10 @@ protected function createNode()
             }
 
             if ($this->default) {
+                if (!\is_array($this->defaultValue)) {
```

Comment:
only for 5? missing deprecation in 4.4?

Reply:
I just moved the check from `PrototypedArrayNode.php::setDefaultValue` here so the typhint in `setDefaultValue` can be added. The `$this->defaultValue` comes from a generic setter that we cannot use types for.